src/app/entry/entry-file-tab/state/entry-file-tab.service.ts

```diff
@@ -32,6 +32,48 @@ export class EntryFileTabService extends Base {
     super();
   }
 
+  /**
+   * This is going to be problematic in the future because we need to match the Dockstore Validation.TypeEnum with
+   * TRS ToolFile.FileTypeEnum for plugins that can have anything. Right now it's matching only a select few.
+   *
+   * @static
+   * @param {(ToolFile | null)} toolFile  The file that's currently being viewed by the user (from TRS)
+   * @param {(WorkflowVersion | null)} version  The version that's currently being viewed (from Dockstore proprietary endpoint)
+   * @returns {(Object | null)}
+   * @memberof EntryFileTabService
+   */
+  public static getValidationMessageObject(toolFile: ToolFile | null, version: WorkflowVersion | null): Object | null {
+    if (version && version.validations && toolFile) {
+      const validations = version.validations;
+      const supportedValidationTypeEnum: Validation.TypeEnum[] = EntryFileTabService.toolFileToValdationTypeEnums(toolFile.file_type);
+      const foundValidation: Validation = validations.find(validation => supportedValidationTypeEnum.includes(validation.type));
+      if (foundValidation) {
+        const validationObject = JSON.parse(foundValidation.message);
```

Comment:
Is it safe to assume this is always valid JSON? Otherwise an exception will get thrown.